radis/test/lbl/test_broadening.py

```diff
@@ -938,6 +938,57 @@ def test_noneq_continuum(plot=False, verbose=2, warnings=True, *args, **kwargs):
     assert res < 5.2e-6
 
 
+def test_broadening_chunksize_eq(verbose=True, plot=False, *args, **kwargs):
+    """
+    Test an equilibrium spectrum with and without chunksize,
+    using different optimization strategies
+    and ensure that the residual is very small.
+
+    """
+    sf = SpectrumFactory(
+        wavelength_min=4000,
+        wavelength_max=4500,
+        cutoff=1e-27,
+        pressure=1,
+        isotope="1,2",
+        truncation=5,
+        neighbour_lines=5,
+        path_length=0.1,
+        mole_fraction=1e-3,
+        medium="vacuum",
+        optimization=None,
+        chunksize=None,
+        wstep=0.001,
+        verbose=False,
+    )
+    sf.load_databank("HITEMP-CO")
+    sf.params.broadening_method = "convolve"
+    Tgas = 2000
+    chunk_loop = 1000000
+    plot_diff_list = []
+    # Testing all 6 cases of chunksize with optimization
+    for chunksize in [None, chunk_loop]:
+        sf.misc["chunksize"] = chunksize  # Setting chunksize
+        s_new = sf.eq_spectrum(Tgas=Tgas)
+
+        for optimization in [None, "simple", "min-RMS"]:
+            sf.params["optimization"] = optimization  # Setting optimization
+            s = sf.eq_spectrum(Tgas=Tgas)
+            # Residual calculation
+            res = get_residual(s, s_new, "abscoeff")
+            assert res < 10e-5
+            print(
+                "Res for chunksize = {0}, optimization = {1} : {2}".format(
+                    chunksize, optimization, res
+                )
+            )
+            if plot:
+                plot_diff_list.append([s, s_new])
```

Comment:
suggestion : if plot, plot directly there instead of delaying to the end ?

Reply:
My thinking was that, if we plot after every calculation process it will hinder the rest of the calculations since the user will have to close the window each time, and if we plot at the end then they'll appear in succession, and it'll be good for comparison as well. But I can shift it above as well, if needed😄

Reply:
As I always run tests in interactive mode i don't have this problem

Reply:
Alright, then I'll move the plot lines to the loop. Thanks for the suggestion!